clorinde/src/codegen/client.rs

```diff
@@ -1,57 +1,69 @@
-use std::fmt::Write;
+use quote::quote;
 
-use codegen_template::code;
-
-use super::{vfs::Vfs, DependencyAnalysis, WARNING};
+use super::{vfs::Vfs, DependencyAnalysis};
 use crate::config::Config;
 
-pub(crate) fn gen_lib(dependency_analysis: &DependencyAnalysis, config: &Config) -> String {
-    let mut w = String::new();
-    code!(w => $WARNING
+pub(crate) fn gen_lib(
+    dependency_analysis: &DependencyAnalysis,
+    config: &Config,
+) -> proc_macro2::TokenStream {
+    let base_tokens = quote! {
         #[allow(clippy::all, clippy::pedantic)]
         #[allow(unused_variables)]
         #[allow(unused_imports)]
         #[allow(dead_code)]
         pub mod types;
+
         #[allow(clippy::all, clippy::pedantic)]
         #[allow(unused_variables)]
         #[allow(unused_imports)]
         #[allow(dead_code)]
         pub mod queries;
+
         pub mod client;
 
         mod array_iterator;
         mod domain;
         mod type_traits;
         mod utils;
 
-        pub (crate) use utils::slice_iter;
+        pub(crate) use utils::slice_iter;
 
         pub use array_iterator::ArrayIterator;
         pub use domain::{Domain, DomainArray};
         pub use type_traits::{ArraySql, BytesSql, IterSql, StringSql};
-    );
+    };
 
-    if config.r#async {
-        code!(w =>
+    let db_imports = if config.r#async {
+        quote! {
             #[cfg(feature = "deadpool")]
             pub use deadpool_postgres;
+
             #[cfg(any(feature = "deadpool", feature = "wasm-async"))]
             pub use tokio_postgres;
+
             #[cfg(not(any(feature = "deadpool", feature = "wasm-async")))]
             pub use postgres;
-        );
+        }
     } else {
-        code!(w =>
+        quote! {
             pub use postgres;
-        );
-    }
+        }
+    };
 
-    if dependency_analysis.json {
-        code!(w => pub use type_traits::JsonSql; )
-    }
+    let json_imports = if dependency_analysis.json {
```

Comment:
I think you can do this instead to minimize boilerplate:

```rust
dependency_analysis.json.map(|_| quote! { pub use type_traits::JsonSql; })
```

Reply:
that doesn't really work, closest I could get to it was this:
```rust
    let json_imports = dependency_analysis
        .json
        .then_some(quote! {
            pub use type_traits::JsonSql;
        })
        .unwrap_or_else(|| quote!());
```
do you prefer that ?